pkg/proxy/iptables/proxier.go

```diff
@@ -204,11 +204,8 @@ func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
 		encounteredError = true
 	}
 
-	args = []string{"-m", "comment", "--comment", "kubernetes service traffic requiring SNAT", "-m", "mark", "--mark", iptablesMasqueradeMark, "-j", "MASQUERADE"}
-	if err := ipt.DeleteRule(utiliptables.TableNAT, utiliptables.ChainPostrouting, args...); err != nil {
-		glog.Errorf("Error removing pure-iptables proxy rule: %v", err)
-		encounteredError = true
-	}
+	// remove all masquerade rules from the POSTROUTING chain.
+	encounteredError = ensureMasqueradeRuleOnly(ipt, "") || encounteredError
```

Comment:
Ohh, good catch.  I think this should be rejiggered to have a single jump rule in POSTROUTING to a KUBE-MASQ chain.  Then cleanup is a single unqualified delete from POSTROUTING and a flush and remove of KUBE-MASQ.

If I am asking too much, say so, and we can do that as a followup